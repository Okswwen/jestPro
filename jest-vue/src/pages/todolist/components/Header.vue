<template>
  <div class="header">
    <div class="h_wrap">
      <div class="todoList">todoList</div>
      <input
        type="text"
        data-test="input"
        v-model="iptValue"
        placeholder="doing something"
        @keyup.enter="addToDoItem"
        />
    </div>
  </div>
</template>

<script>
export default {
  name: 'Header',
  props: {
    msg: String
  },
  data () {
    return {
      iptValue: ''
    }
  },
  methods: {
    addToDoItem () {
      if (this.iptValue) {
        this.$emit('add', this.iptValue)
        this.iptValue = ''
      }
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
.header {
  position: relative;
  height: 80px;
  line-height: 80px;
  background-color: #999;
  .h_wrap {
    width: 800px;
    margin: 0 auto;
    overflow: hidden;
  }
  .todoList {
    float: left;
    font-size: 24px;
    color: #fff;
  }
  input {
    margin-left: 60px;
    width: 300px;
    height: 44px;
    border: 1px solid #666;
    padding-left: 15px;
    border-radius: 8px;
  }
}

</style>
