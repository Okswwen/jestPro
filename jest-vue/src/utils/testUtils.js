const findTestWrapper = (warpper, tag) => {
  return warpper.findAll(`[data-test='${tag}']`)
}

export default {
  findTestWrapper
}
