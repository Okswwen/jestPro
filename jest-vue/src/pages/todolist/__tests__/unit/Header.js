import { shallowMount } from '@vue/test-utils'
import Header from '../../components/Header.vue'
import utils from '../../../../utils/testUtils'

it('Header 样式发生变化，做出提示 ', () => {
  const wrapper = shallowMount(Header)
  expect(wrapper).toMatchSnapshot()
})

it('Header 包含一个 input 框 ', () => {
  const wrapper = shallowMount(Header)
  const input = utils.findTestWrapper(wrapper, 'input')
  // wrapper.find('[data-test = "input"]')
  expect(input.exists()).toBe(true)
})

it('Header 包含一个 input 框,初始值为空 ', () => {
  const wrapper = shallowMount(Header)
  const iptValue = wrapper.vm.$data.iptValue
  expect(iptValue).toBe('')
})

it('Header 中input 框,value变化时，数据跟着改变 ', () => {
  const wrapper = shallowMount(Header)
  const input = wrapper.find('[data-test = "input"]')
  input.setValue('steven')
  const iptValue = wrapper.vm.$data.iptValue
  expect(iptValue).toBe('steven')
})

it('Header 中input 框,无内容，回车无反应 ', () => {
  const wrapper = shallowMount(Header)
  const input = wrapper.find('[data-test = "input"]')
  input.setValue('')
  input.trigger('keyup.enter')

  expect(wrapper.emitted().add).toBeFalsy()
})

it('Header 中input 框,有内容，向外触发事件,同时清除iptValue的值 ', () => {
  const wrapper = shallowMount(Header)
  const input = wrapper.find('[data-test = "input"]')
  const iptValue = wrapper.vm.$data.iptValue
  input.setValue('steven')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeTruthy()
  expect(iptValue).toBe('')
})
