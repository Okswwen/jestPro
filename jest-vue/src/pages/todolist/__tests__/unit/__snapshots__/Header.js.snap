// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式发生变化，做出提示  1`] = `
<div class="header">
  <div class="h_wrap">
    <div class="todoList">todoList</div> <input type="text" data-test="input" placeholder="doing something">
  </div>
</div>
`;
